lib/jormungandr/src/Cardano/Wallet/Binary/Jormungandr.hs

```diff
@@ -106,21 +101,129 @@ getBlock = do
+data Message
+    = Initial [ConfigParam]
+    | OldUtxoDeclaration TODO
+    | Transaction TODO
+    | Certificate TODO
+    | UpdateProposal SignedUpdateProposal
+    | UpdateVote SignedVote
+    | UnimplementedMessage Int -- For development. Remove later.
+    deriving (Eq, Show)
+
 getMessage :: Get Message
 getMessage = do
     size <- fromIntegral <$> getWord16be
     contentType <- fromIntegral <$> getWord8
     let remaining = size - 1
     let unimpl = skip remaining >> return (UnimplementedMessage contentType)
     isolate remaining $ case contentType of
-        0 -> unimpl
+        0 -> Initial <$> getInitial
         1 -> unimpl
         2 -> unimpl
         3 -> unimpl
         4 -> unimpl
         5 -> unimpl
         other -> fail $ "Unexpected content type tag " ++ show other
 
+getInitial :: Get [ConfigParam]
+getInitial = do
+    len <- fromIntegral <$> getWord16be
+    replicateM len getConfigParam
+
+{-------------------------------------------------------------------------------
+                            Config Parameters
+-------------------------------------------------------------------------------}
+
+data ConfigParam
+    -- Seconds elapsed since 1-Jan-1970 (unix time)
+    = Block0Date Word64
+    | ConfigDiscrimination Discrimination
+    | ConsensusVersion Word16 -- ?
+    | SlotsPerEpoch Word32
+    | SlotDuration Word8
+    | EpochStabilityDepth Word32
+    | ConsensusGenesisPraosActiveSlotsCoeff Milli
+    | MaxNumberOfTransactionsPerBlock Word32
+    | BftSlotsRatio Milli
+    | AddBftLeader LeaderId
+    | RemoveBftLeader LeaderId
+    | AllowAccountCreation Bool
+    | ConfigLinearFee LinearFee
+    | ProposalExpiration Word32
+    deriving (Eq, Show)
+
+getConfigParam :: Get ConfigParam
+getConfigParam = do
+    -- The tag and the size/length of the config param is stored in a single
+    -- @Word16@.
+    --
+    -- 6 least-significant bits: length
+    -- 12 most-significant bits: tag
+    taglen <- getWord16be
+    let tag = taglen `shift` (-6)
+    let len = fromIntegral $ taglen .&. (63) -- 0b111111
+
+    isolate len $ case tag of
+        1 -> ConfigDiscrimination <$> getDiscrimination
+        2 -> Block0Date <$> getWord64be
+        3 -> ConsensusVersion <$> getWord16be -- ?
+        4 -> SlotsPerEpoch <$> getWord32be
+        5 -> SlotDuration <$> getWord8
+        6 -> EpochStabilityDepth <$> getWord32be
+        8 -> ConsensusGenesisPraosActiveSlotsCoeff <$> getMilli
+        9 -> MaxNumberOfTransactionsPerBlock <$> getWord32be
+        10 -> BftSlotsRatio <$> getMilli
+        11 -> AddBftLeader <$> getLeaderId
+        12 -> RemoveBftLeader <$> getLeaderId
+        13 -> AllowAccountCreation <$> getBool
+        14 -> ConfigLinearFee <$> getLinearFee
+        15 -> ProposalExpiration <$> getWord32be
+        a -> fail $ "Invalid config param with tag " ++ show a
+
+data Discrimination = Production | Test
+    deriving (Eq, Show)
```

Comment:
That should basically match the `Cardano.Wallet.Environment.Jormungandr`. There's no more protocol magic with Jormungandr but instead, only two levels of discrimination.